pinot-common/src/main/java/org/apache/pinot/common/request/context/predicate/RegexpLikePredicate.java

```diff
@@ -48,6 +50,13 @@ public String getValue() {
     return _value;
   }
 
+  public Pattern getPattern() {
+    if (_pattern == null) {
```

Comment:
Do this lazily because it's an overhead when an FST index is available

Reply:
The pattern can be accessed by multiple threads, so might be better to make it `volatile` or make it atomic swap?

Reply:
It doesn't matter, the worst that can happen is it gets compiled more than once because the operation is idempotent.